src/coreclr/src/gc/gc.cpp

```diff
@@ -5927,6 +5965,19 @@ bool gc_heap::virtual_decommit (void* address, size_t size, gc_oh_num oh, int h_
 void gc_heap::virtual_free (void* add, size_t allocated_size, heap_segment* sg)
 {
     assert(!heap_hard_limit);
+#ifdef VERIFY_COMMITTED_BY_OH
+    if (sg)
+    {
+        size_t committed_size = (uint8_t*)heap_segment_committed (sg)-(uint8_t*)sg;
+        gc_oh_num oh = heap_segment_oh (sg);
+#ifdef MULTIPLE_HEAPS
+        heap_segment_heap (sg)->committed_by_oh_per_heap[oh] -= committed_size;
```

Comment:
>heap_segment_heap (sg)->committed_by_oh_per_heap[oh] -= committed_size; [](start = 8, length = 71)

virtual_free can be called from things that would call virtual_commit with -1 as the heap number.

Reply:
This line is conditioned on `sg != NULL`, meaning only the call from `gc_heap::release_segment` will reach here.